modules/ding_debt_easy/lib/Nets/Easy/Client.inc

```diff
@@ -0,0 +1,220 @@
+<?php
+
+namespace Nets\Easy;
+
+use Nets\Easy\Exception\PaymentCommunicationException;
+use Nets\Easy\Exception\PaymentException;
+
+/**
+ * Client to send request to the payment gateway at Nets Easy.
+ */
+class Client {
+  private $secretKey;
+  private $checkoutKey;
+  private $apiUrl;
+  private $apiPaymentsPath = '/v1/payments/';
+
+  public function __construct($secretKey, $checkoutKey, $apiUrl = 'https://api.dibspayment.eu/') {
+    $this->secretKey = $secretKey;
+    $this->apiUrl = $apiUrl;
+
+    // The checkout key is only used during embedded checkout process
+    // (not implemented yet).
+    $this->checkoutKey = $checkoutKey;
+  }
+
+  /**
+   * Create payment at the gateway.
+   *
+   * @param Checkout $checkout
+   *   Checkout order representation (the request to send to the gateway).
+   *
+   * @return array
+   *   Will return the paymentId and if hosted payment type the
+   *   hostedPaymentPageUrl to redirect to.
+   *
+   * @throws \Nets\Easy\Exception\PaymentCommunicationException
+   * @throws \Nets\Easy\Exception\PaymentException
+   */
+  public function createPayment(Checkout $checkout) {
+    $ch = curl_init();
+    curl_setopt_array($ch, [
+      CURLOPT_URL => $this->apiUrl . $this->apiPaymentsPath,
+      CURLOPT_RETURNTRANSFER => true,
+      CURLOPT_ENCODING => "",
+      CURLOPT_MAXREDIRS => 10,
+      CURLOPT_TIMEOUT => 30,
+      CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
+      CURLOPT_CUSTOMREQUEST => "POST",
+      CURLOPT_POSTFIELDS => $checkout->toJson(),
+      CURLOPT_HTTPHEADER => [
+        'Authorization: ' . $this->secretKey,
+        'CommercePlatformTag: DDB CMS',
+        'content-type: application/*+json',
+      ],
+    ]);
```

Comment:
The request from DDF was that it should not be dependent on other libraries.

Reply:
That seems absurd and/or misunderstood. I do not understand the background for such a request. Guzzle is already included in the current project and is available in Drupal 9 as well meaning that the potential for code reuse is not affected by this.

If we want to avoid dependencies we might as well use `drupal_http_request()`. It does not require the `curl` extension to be present.

Reply:
The switch to guzzle have been made...